api/filters_test.go

```diff
@@ -138,6 +138,82 @@ func TestSuccessfulAddFilterBlueprint_PublishedDataset(t *testing.T) {
 	})
 }
 
+func TestSuccessfulPostFilterBlueprint_Flexible(t *testing.T) {
+
+	Convey("Given an flexible filter", t, func() {
+		w := httptest.NewRecorder()
+		filterFlexAPIMock := &apimock.FilterFlexAPIMock{
+			ForwardRequestFunc: func(r *http.Request) (*http.Response, error) {
+				return &http.Response{
+					Body:       io.NopCloser(bytes.NewReader([]byte("test body"))),
+					StatusCode: http.StatusOK,
+				}, nil
+			},
+		}
+		datastoreMock := mock.NewDataStore().Mock
+		datastoreMock.GetFilterFunc = func(ctx context.Context, filterID, etag string) (*models.Filter, error) {
+			return &models.Filter{
+				Type: "flexible",
+			}, nil
+		}
+		filterApi := api.Setup(cfg(), mux.NewRouter(), datastoreMock, &mock.FilterJob{}, &mock.DatasetAPI{}, filterFlexAPIMock)
+
+		Convey("When a post request is made to the submit endpoint", func() {
+			reader := strings.NewReader(`{}`)
+			r := createAuthenticatedRequest("POST", "http://localhost:22100/filters/123456/submit", reader)
+			filterApi.Router.ServeHTTP(w, r)
+
+			Convey("Then the request is not be forwarded to the filter flex api", func() {
+				So(filterFlexAPIMock.ForwardRequestCalls(), ShouldHaveLength, 0)
+			})
+
+			Convey("Then an error is returned", func() {
+				So(w.Code, ShouldEqual, http.StatusBadRequest)
+			})
+		})
+	})
+}
+
+func TestUnSuccessfulPostFilterBlueprint_NonFlexible(t *testing.T) {
```

Comment:
I would have a single function to test this new functionality and use the conveys for the different scenarios? Maybe we could reuse some common setup that way?

Reply:
I initially had the same thought but wanted to keep the current style in the test file of using separate functions.